From 527a7c3d7b9d752e6c0ab83af039e330ef4fbba7 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 09 Oct 2018 19:08:10 +0300
Subject: [PATCH] eeprog: fix the fields' bytes order

The bytes order of multy byte binary fields is "LSB first".
The xxd tool defaults to "MSB first".
Since busybox's xxd does not have the '-e' switch, provide a for loop
to read the values correctly (in reverse order).

Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/board/habanalabs/common/rootfs_overlay/usr/sbin/eeprog b/board/habanalabs/common/rootfs_overlay/usr/sbin/eeprog
index 3046d68..cc9d187 100755
--- a/board/habanalabs/common/rootfs_overlay/usr/sbin/eeprog
+++ b/board/habanalabs/common/rootfs_overlay/usr/sbin/eeprog
@@ -20,7 +20,11 @@
 	NAME=${1}
 	OFFSET=${2}
 	LEN=${3}
-	FIELD=`xxd -p -s${OFFSET} -l${LEN} -c${LEN} ${LAYOUT}`
+	FIELD=""
+	for ((i = 0; i < $LEN; i++)); do
+		CUR_OFF=$((OFFSET + LEN - i - 1))
+		FIELD=${FIELD}`xxd -p -s${CUR_OFF} -l1 -c1 ${LAYOUT}`
+	done
 	echo "${NAME}: ${FIELD}"
 }
 
